lib/ClangImporter/ImportName.h

```diff
@@ -56,12 +56,35 @@ class ImportNameVersion : public RelationalOperationsBase<ImportNameVersion> {
 public:
   /// Map a language version into an import name version.
   static ImportNameVersion fromOptions(const LangOptions &langOpts) {
-    return ImportNameVersion(langOpts.EffectiveLanguageVersion[0]);
+    // We encode the 'rawValue' as just major version numbers with the
+    // exception of '4.2', which is a special minor version that can impact
+    // importing of names.  We treat that with a rawValue of 5, and treat
+    // all major values of 5 or higher as being rawValue = majorversion + 1.
```

Comment:
Since we don't distinguish between Swift 1 and Swift 2, what do you think about moving the raw values down instead of up? It's a trick that will only work once, but even so.